lib/lightning/attempts/prom_ex_plugin.ex

```diff
@@ -78,4 +83,89 @@ defmodule Lightning.Attempts.PromExPlugin do
 
     :telemetry.execute(@stalled_event, %{count: count}, %{})
   end
+
+  defp attempt_performance_metrics(attempt_age_seconds) do
+    Polling.build(
+      :lightning_attempt_queue_metrics,
+      5000,
+      {__MODULE__, :attempt_claim_duration, [attempt_age_seconds]},
+      [
+        last_value(
+          [
+            :lightning,
+            :attempt,
+            :queue,
+            :claim,
+            :average_duration,
+            :milliseconds
+          ],
+          event_name: @average_claim_event,
+          description: "The average time taken before an attempt is claimed",
+          measurement: :average_duration,
+          unit: :millisecond
+        )
+      ]
+    )
+  end
+
+  def attempt_claim_duration(attempt_age_seconds) do
+    trigger_attempt_claim_duration(Process.whereis(Repo), attempt_age_seconds)
+  end
+
+  defp trigger_attempt_claim_duration(nil, _attempt_age_seconds) do
+    nil
+  end
+
+  defp trigger_attempt_claim_duration(repo_pid, attempt_age_seconds) do
+    check_repo_state(repo_pid)
+
+    average_duration =
+      calculate_average_claim_duration(DateTime.utc_now(), attempt_age_seconds)
+
+    :telemetry.execute(
+      @average_claim_event,
+      %{average_duration: average_duration},
+      %{}
+    )
+  end
+
+  def calculate_average_claim_duration(reference_time, attempt_age_seconds) do
+    threshold_time = reference_time |> DateTime.add(-attempt_age_seconds)
+
+    query =
+      from a in Attempt,
+        where: a.state == :available,
+        or_where: a.state != :available and a.inserted_at > ^threshold_time
+
+    query
+    |> Repo.all()
+    |> Enum.reduce({0, 0}, fn attempt, {sum, count} ->
+      {sum + claim_duration(attempt, reference_time), count + 1}
+    end)
+    |> average()
+  end
+
+  defp claim_duration(%Attempt{state: :available} = attempt, reference_time) do
+    DateTime.diff(reference_time, attempt.inserted_at, :millisecond)
+  end
+
+  defp claim_duration(attempt, _reference_time) do
+    DateTime.diff(attempt.claimed_at, attempt.inserted_at, :millisecond)
+  end
+
+  defp average({_sum, 0}) do
+    0.0
+  end
+
+  defp average({sum, count}) do
+    (sum / count) |> Float.round(0)
```

Comment:
as the unit is `millisecond` if we can would use integer format to replace the decimal with 0 digits: `trunc(sum / count)`